cdap-api/src/main/java/co/cask/cdap/api/messaging/MessageFetcher.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.messaging;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.dataset.lib.CloseableIterator;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * Provides access to fetch messages from the transactional messaging system.
```

Comment:
The `MessageFetcher` interface is for fetching messages from TMS. Depending on the context, it can be doing transactional fetch or non-transactional fetch. That has been documented at the `MessagingContext` methods, because those are the methods that the user will use first to acquire instance of `MessageFetcher` or `MessagePublisher`.